core/exec/src/mill/exec/Execution.scala

```diff
@@ -188,7 +195,22 @@ private[mill] case class Execution(
                   exclusive
                 )
 
-                if (failFast && res.newResults.values.exists(_.asSuccess.isEmpty))
+                // After execution, only count new failures if there were no upstream failures
+                val hasUpstreamFailures = upstreamResults.values.exists(r => r.asFailing.isDefined)
```

Comment:
Do we need this branch? If there are upstream failures, the current task should never return a failure, and should be `Skipped`